docs/components/index.md

```diff
@@ -1,26 +1,156 @@
 <script setup>
-const components = import.meta.glob('./*/data.json', { eager: true })
-const baseUrl = import.meta.env.BASE_URL
+import { ref, computed } from 'vue';
 
-const componentData = Object.keys(components).map(path => {
- return ({
-  ...components[path].default, // component data from JSON
+// Import all JSON data
+const components = import.meta.glob('./*/data.json', { eager: true });
+const baseUrl = import.meta.env.BASE_URL;
+
+// Map JSON data to include href and correct paths
+const componentData = Object.keys(components).map(path => ({
+  ...components[path].default,
   href: `${baseUrl}components/${path.replace('/data.json', '')}`
-})});
+}));
+
+// Reactive variables for query and selected frameworks
+const query = ref('');
+const selectedFrameworks = ref([]);
+
+// Extract unique framework names
+const frameworkNames = computed(() => {
+  const names = new Set();
+  componentData.forEach(component => {
+    component.frameworks.forEach(framework => {
+      names.add(framework.name);
+    });
+  });
+  return Array.from(names);
+});
+
+// Computed property to filter components based on query and selected frameworks
+const filteredComponents = computed(() => {
+  const lowerQuery = query.value.toLowerCase();
+  const frameworks = selectedFrameworks.value;
+
+  return componentData.filter(component => {
+    const matchesQuery = component.title.toLowerCase().includes(lowerQuery);
+    const matchesFramework = frameworks.length === 0 ||
+      component.frameworks.some(framework => frameworks.includes(framework.name));
+    return matchesQuery && matchesFramework;
+  });
+});
 
+// Function to handle button clicks
+const toggleFramework = (framework) => {
+  if (selectedFrameworks.value.includes(framework)) {
+    selectedFrameworks.value = selectedFrameworks.value.filter(f => f !== framework);
+  } else {
+    selectedFrameworks.value.push(framework);
+  }
+};
+
+// Function to handle "All" button click
+const showAll = () => {
+  selectedFrameworks.value = [];
+};
 </script>
 
+<style>
+  /* remove these to custom.css and replace with warp-doc color tokens */
+  :root{
+    --w-border: #999;
+    --w-border-focus: #00F;
+    --w-button-bg: #eee;
+    --w-button-bg-active: #333;
+    --w-text: #333;
+    --w-text-inverted: #eee;
+    --w-border-radius: 4px;
+  }
+  /* set up as custom default styling somewhere else, default.css perhaps?  */
```

Comment:
Is it that or creating an InputField and Button component with scoped styles? If so, which way do you think it would make sense for us to go for here?

Reply:
Hmm.. Good question. I dunno if it is worth the hassle to make form elements as components for us. They are sort of already components out of the box. We could keep them as regular html elements until we see a point in doing them as components I think. We do need to keep an extra eye on accessibility though, but thats goes no matter if they are components or not :)

Reply:
I see! I'm wondering how to best ensure we remember that these components already were styled here. Maybe as you suggested in the comment, we could already move these styles to some default.css? Wdyt?

Reply:
They will not be styled here, Im just keeping them here temporary to not mess up other things before we get the actual tokens we are gonna use.